Added types/pico8parse

PictElm opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

andrewbranch commented on types/pico8parse/lib/ast.d.ts:
```diff
@@ -0,0 +1,337 @@
+export interface Base<TType extends string> {
```
This file implies that you can import from `"pico8parse/lib/ast"`, which does not actually exist on npm, so unfortunately everything will have to go in the index file.

andrewbranch commented on types/pico8parse/index.d.ts:
```diff
@@ -0,0 +1,129 @@
+export class SyntaxError extends Error {}
+
+declare const parser: Parser;
+export default parser;
```
This would be written better as a UMD definition, since [it’s published that way](https://unpkg.com/browse/pico8parse@0.3.1001/luaparse.js).

PictElm commented:
Thanks for the quick reaction ! Will do appropriate changes.

PictElm commented:
As requested, removed the extraneous `./lib/ast` and changed to UMD definition (I think..? I got confused a lot and had to use a `tslint:disable:export-just-namespace`)

andrewbranch commented:
Yeah, so basically you can remove the `Pico8Parser` namespace wrapper and add an explicit `export` modifier to everything in it, and the linter should stop complaining.

PictElm commented:
Oh, I see, it does work ! I through I tried that, but I most likely had something off. Thanks !

PictElm commented:
Ready to merge
